WooCommerce/UITestsFoundation/Screens/Login/PasswordScreen.swift

```diff
@@ -33,16 +33,15 @@ public final class PasswordScreen: ScreenObject {
         _ = try tryProceed(password: password)
     }
 
-    public func tryProceed(password: String) throws -> PasswordScreen {
+    public func tryProceed(password: String) throws {
         passwordField.enterText(text: password)
         continueButton.tap()
         if continueButton.exists && !continueButton.isHittable {
             waitFor(element: continueButton, predicate: "isEnabled == true")
         }
-
-        return self
     }
```

Comment:
🤔 I worry  breaking the pattern that allows us to chain method calls might result in confusion when looking at the tests (e.g. someone asking why do all methods return `self` but this one doesn't?) as well as tests that don't look homogeneous.

> Found out that after entering the password, the method returns self and one of the checks for that in the Continue Button. Due to the slowness in CI, at that point, the password screen is no longer displayed, and the test is already on the My Store screen, so it will fail as the Continue button is no longer on that screen.

Could  the issue be related with the `if continueButton.exists...` check right after `tap()`? Why do we check for it to be hittable _after_ attempting to hit it?

Reply:
it looks like the existing `tryProceed(password: String)` returns 2 different values depending on the scenario, when entering a valid password, it should return the My Store screen, if it's an invalid password, it should return the Password screen which was the reason it has the check for the continue button after tapping on it. 

i've seen this particular step failing quite a bit recently in [a testing branch for parallel testing](https://github.com/woocommerce/woocommerce-ios/pull/9270) across multiple tests (it's being used during `setUp()` too) so i think a better solution is to break it and get it to return the correct screen depending on the situation instead of using one for both (returning password screen when expecting my store screen is incorrect). 

as to why it's been passing most the time even though incorrect, i think depends a lot on how fast/slow the test is (it's a fast check but after turning on parallel testing, it became slower so the issue became more obvious). 

updated in https://github.com/woocommerce/woocommerce-ios/pull/9560/commits/d5b92da40616e3884d97691a7290a4447f0a58bf